Documentation for AIA (#26940)
keycloak/keycloak: docs/documentation/server_admin/topics/authentication/flows.adoc

one of the specified levels. If it is not able to return one of the specified levels (For example if the requested level is unknown or bigger than configured conditions
in the authentication flow), then {project_name} will throw an error.

[[_registration-rc-client-flows]]
==== Registration or Reset credentials requested by client

> Usually when user is redirected to the {project_name} from client application, the `browser` flow is triggered. This flow may allow user to <<con-user-registration_{context}, register>> in case

Reviewer: ```suggestion
Usually when the user is redirected to the {project_name} from client application, the `browser` flow is triggered. This flow may allow the user to <<con-user-registration_{context}, register>> in case
```

> that realm registration is enabled and user clicks to the link `Register` on the login screen. Also in case that <<enabling-forgot-password, Forget password>> is enabled for the realm, user can

Reviewer: ```suggestion
that realm registration is enabled and the user clicks `Register` link on the login screen. Also, if  <<enabling-forgot-password, Forget password>> is enabled for the realm, the user can
```

> click on the link `Forget password` on the login screen, which will trigger `Reset credentials` flow where user can reset his credentials after confirmation of his email address.

Reviewer: ```suggestion
click `Forget password` lon the login screen, which triggers the `Reset credentials` flow where users can reset credentials after email address confirmation.
```

> Sometimes it can be useful for client application to directly redirect user to the *Registration* screen or to *Reset credentials* flow. This will have exactly same behaviour like when manually

Reviewer: ```suggestion
Sometimes it can be useful for the client application to directly redirect the user to the *Registration* screen or to the *Reset credentials* flow. This resulting action will match the action of when the 
```

> user clicks *Register* or *Forget password* links on the normal login screen. Automatically redirect to registration or reset-credentials screen can be done as follows:

Reviewer: ```suggestion
user clicks *Register* or *Forget password* on the normal login screen. Automatic redirect to the registration or reset-credentials screen can be done as follows:
```

> * When client wants user to be redirected directly to the registration, OIDC client should replace very last snippet from the OIDC login URL path (`/auth`) with `/registrations` . So the full URL

Reviewer: ```suggestion
* When the client wants the user to be redirected directly to the registration, the OIDC client should replace the very last snippet from the OIDC login URL path (`/auth`) with `/registrations` . So the full URL
```

> might be something like `https://your-keycloak-host/realms/your_realm/protocol/openid-connect/registrations`.

Reviewer: ```suggestion
might be similar to the following: `https://your-keycloak-host/realms/your_realm/protocol/openid-connect/registrations`.
```

> * When client wants user to be redirected directly to the `Reset credentials` flow, OIDC client should replace very last snippet from the OIDC login URL path (`/auth`) with `/forgot-credentials` .

Reviewer: ```suggestion
* When the client wants a user to be redirected directly to the `Reset credentials` flow, the OIDC client should replace the very last snippet from the OIDC login URL path (`/auth`) with `/forgot-credentials` .
```

> WARNING: The steps above are only officially supported way how client can directly request registration or reset-credentials flow. For security

Reviewer: ```suggestion
WARNING: The steps above are only officially supported way how client can directly request registration or reset-credentials flow. For security
```
```suggestion
WARNING: The preceding steps are the only supported method to how a client can directly the request registration or reset-credentials flow. For security
```

Reviewer: I removed "officially" because it sounds as if there "unofficially" supported methods.

> WARNING: The steps above are only officially supported way how client can directly request registration or reset-credentials flow. For security

Reviewer: ```suggestion
WARNING: The preceding steps are the only supported method for a client to directly request a registration or reset-credentials flow. For security
```

> purposes, it is not supported and recommended for client applications to bypass OIDC/SAML flows and directly redirect to other {project_name} endpoints (like for instance endpoints under

Reviewer: ```suggestion
purposes, it is not supported and recommended for client applications to bypass OIDC/SAML flows and directly redirect to other {project_name} endpoints (such as for instance endpoints under
```

Reviewer: We prefer "such as" rather than "like" in technical documentation
